src/Query/Traits/JoinComponentCompiler.php

```diff
@@ -55,11 +64,11 @@ protected function compileJoinComponent(Builder $query, JoinClause $join): strin
      *
      * @throws GrammarException
      */
-    private function verifyJoin(JoinClause $joinClause)
+    private function verifyJoin(JoinClause $joinClause): void
     {
         if (
             is_null($joinClause->getTable()) ||
-            is_null($joinClause->getUsing())
+            (is_null($joinClause->getUsing()) && is_null($joinClause->getOnClauses()))
```

Comment:
Accordingly to the clickhouse documentation join cannot have both `on` and `using` clauses, so we need to check if the join doesn't have them both.